sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -92,10 +92,14 @@ export class TestProxyHttpClient {
    * redirectRequest updates the request in record and playback modes to hit the proxy-tool with appropriate headers.
    * Works for both core-v1 and core-v2
    */
-  redirectRequest(request: WebResourceLike | PipelineRequest) {
+  redirectRequest(request: WebResourceLike | PipelineRequest): typeof request {
```

Comment:
Looks like this function is updating the request in place. If so, does it even have to return anything?

Reply:
That's an interesting point... looking into it, there are definitely places (e.g. [in the tests here](https://github.com/Azure/azure-sdk-for-js/blob/49189e9a9f1f179c02dd4df3b44c5688c79d2034/sdk/test-utils/recorder-new/test/testProxyClient.spec.ts#L52)) where it's incorrectly assumed that it's __not__ updating the request in place.

Reply:
Hmmm, can you check if we miss anything if we don't return, @timovv ?

Reply:
In most places it would be fine not to return anything, it's just that there are a couple of spots like in the test I linked above where it assumes that the request isn't updated in place.

Reply:
Tests are fine, they can be updated. 🥲
If it's not too much work, can you update the method to not return and make sure nothing is broken? and update the tests accordingly.

Reply:
TBH either is fine.
For the fresh object to be returned, create a new type if necessary. :)

Reply:
It would have to be the same type, I believe, due to the way the request is used afterwards. Just not returning will definitely be easier, so will go with that.